src/Uno.UI/UI/Xaml/Media/Imaging/RenderTargetBitmap.cs

```diff
@@ -142,36 +221,57 @@ public IAsyncOperation<IBuffer> GetPixelsAsync()
 				{
 					return Task.FromResult<IBuffer>(new Buffer(Array.Empty<byte>()));
 				}
+
+#if !__ANDROID__
+				unsafe
+				{
+					var mem = new UnmanagedMemoryManager<byte>((byte*)_buffer.Value.Pointer.ToPointer(), _bufferSize);
+					return Task.FromResult<IBuffer>(new Buffer(mem.Memory.Slice(0, _bufferSize)));
+				}
+#else
 				return Task.FromResult<IBuffer>(new Buffer(_buffer.AsMemory().Slice(0, _bufferSize)));
+#endif
 			});
 
 #if NOT_IMPLEMENTED
-		private (int ByteCount, int Width, int Height) RenderAsBgra8_Premul(UIElement element, ref byte[]? buffer, Size? scaledSize = null)
+		private (int ByteCount, int Width, int Height) RenderAsBgra8_Premul(UIElement element, ref UnmanagedArrayOfBytes? buffer, Size? scaledSize = null)
 			=> throw new NotImplementedException("RenderTargetBitmap is not supported on this platform.");
 #endif
 
-		void IDisposable.Dispose()
+#if !__ANDROID__
+		~RenderTargetBitmap()
 		{
-			if (_buffer is not null)
+			if (_buffer.HasValue)
 			{
-				ArrayPool<byte>.Shared.Return(_buffer);
+				_buffer.Value.Dispose();
 			}
```

Comment:
Can't the buffer handle that by itself?

Reply:
I don't see a way for that :'(

Reply:
Maybe switch `UnmanagedArrayOfBytes` to class instead of struct and put that code in UnmanagedArrayOfBytes's finalizer?